c7n/actions/webhook.py

```diff
@@ -0,0 +1,154 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+import certifi
+import jmespath
+import urllib3
+from six.moves.urllib import parse
+
+from c7n import utils
+from .core import EventAction
+
+
+class Webhook(EventAction):
+    """Calls a webhook with optional parameters and body
+       populated from JMESPath queries.
+
+        .. code-block:: yaml
+
+          policies:
+            - name: call-webhook
+              resource: ec2
+              description: |
+                Call webhook with list of resource groups
+              actions:
+               - type: webhook
+                 url: http://foo.com
+                 query-params:
+                    resource_name: resource.name
+                    policy_name: policy.name
+    """
+
+    schema = utils.type_schema(
+        'webhook',
+        required=['url'],
+        **{
+            'url': {'type': 'string'},
+            'body': {'type': 'string'},
+            'batch': {'type': 'boolean'},
+            'batch-size': {'type': 'number'},
+            'method': {'type': 'string', 'enum': ['PUT', 'POST', 'GET', 'PATCH', 'DELETE']},
+            'query-params': {
+                "type": "object",
+                "additionalProperties": {
+                    "type": "string",
+                    "description": "query string values"
+                }
+            },
+            'headers': {
+                "type": "object",
+                "additionalProperties": {
+                    "type": "string",
+                    "description": "header values"
+                }
+            }
+        }
+    )
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super(Webhook, self).__init__(data, manager, log_dir)
+        self.http = None
+        self.url = self.data.get('url')
+        self.body = self.data.get('body')
+        self.batch = self.data.get('batch', False)
+        self.batch_size = self.data.get('batch-size', 500)
+        self.query_params = self.data.get('query-params', {})
+        self.headers = self.data.get('headers', {})
+        self.method = self.data.get('method', 'POST')
+        self.lookup_data = {
+            'account_id': self.manager.config.account_id,
+            'region': self.manager.config.region,
+            'execution_id': self.manager.ctx.execution_id,
+            'execution_start': self.manager.ctx.start_time,
+            'policy': self.manager.data
+        }
+
+    def process(self, resources, event=None):
+        self.http = urllib3.PoolManager(
+            cert_reqs='CERT_REQUIRED',
+            ca_certs=certifi.where())
+
+        if self.batch:
```

Comment:
Would be great to run it in parallel. We can only create an issue to add it in the future for now

Reply:
Since this was a core C7N action I referenced the two similar actions (Notify and Invoke) and they did not do this, so I kept it the same.  I don't really have an opinion.  Do you think there are common webhook scenarios where it would benefit?  I assume you'd have to be running `batch: false` and your webhook would need to respond slowly but also handle parallel requests well.